src/Microsoft.DotNet.Build.Tasks/Targets/partialfacades.targets

```diff
@@ -0,0 +1,141 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  
+  <UsingTask TaskName="ResolveNuGetPackageAssets" AssemblyFile="$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll"/>
+
+  <PropertyGroup>
+    <GenFacadesNuGetName>Microsoft.DotNet.BuildTools.ApiTools.1.0.0-prerelease</GenFacadesNuGetName>
+    <GenFacadesPath>$(PackagesDir)$(GenFacadesNuGetName)\tools\GenFacades.exe</GenFacadesPath>
+  </PropertyGroup>
+
+  <!-- Default properties for partial facade projects -->
+  <PropertyGroup Condition="'$(IsPartialFacadeAssembly)' == 'true'">
+    <PostFilterNugetReferences>true</PostFilterNugetReferences>
+    <IgnoreArchitectureMismatches>true</IgnoreArchitectureMismatches>
+  </PropertyGroup>
+
+  <!-- It seems this needs to be set here, and cannot be set in individual project files -->
+  <PropertyGroup Condition="'$(IgnoreArchitectureMismatches)' == 'true'">
+    <!-- Needed because we are using mscorlib.dll from the CoreCLR package, rather than a true reference assembly -->
+    <ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>None</ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>
+  </PropertyGroup>
+
+  <!-- PostFilterNuGetReferences
+       Filters the set of references generated from the ResolveNugetPackages target. This is a default target for partial facade
+         assemblies, and can be optionally used in other projects as well, by setting PostFilterNugetReferences=true. The target
+         contains some logic for removing well-known assemblies from the "Microsoft.DotNet.CoreCLR" package, except for "mscorlib.dll".
+  -->
+  <Target Name="PostFilterNugetReferences"
+    Condition="'$(PostFilterNugetReferences)' == 'true'"
+    AfterTargets="ResolveNugetPackages">
+
+    <!-- Well-known default removals -->
+    <ItemGroup>
+      <!-- Only take mscorlib from Microsoft.DotNet.CoreCLR. -->
+      <ReferenceRemovals Include="@(Reference)" Condition="('%(Reference.NuGetPackageName)' == 'Microsoft.DotNet.CoreCLR' or '%(Reference.NuGetPackageId)' == 'Microsoft.DotNet.CoreCLR') and '%(FileName)' != 'mscorlib'"/>
+      <!-- We don't want any of these from Microsoft.DotNet.CoreCLR, either. -->
+      <NoneRemovals Include="@(Reference)" Condition="'%(Reference.NuGetPackageName)' == 'Microsoft.DotNet.CoreCLR'"/>
+    </ItemGroup>
+
+    <ItemGroup>
+      <Reference Remove="@(ReferenceRemovals)" />
+      <None Remove="@(NoneRemovals)" />
+    </ItemGroup> 
+  </Target>
+
+  <!-- FindPartialFacadeContractReference
+         Uses the special partial-facade project.json file to restore the NuGet package containing the partial facade's contract assembly.
+         Filters the list of references returned by the ResolveNuGetPackageAssets task to just the contract assembly and stores it.
+  -->
```

Comment:
Yes based on @ericstj mail that would appear to be the case. In which case can we try to hide this? I assume this project.json file only contains the contract name/version correct? Can we generate that into the intermediates directory as part of the build and then restore from there instead of requiring all these projects to have this extra set of stuff?